lib/spack/spack/util/executable.py

```diff
@@ -141,15 +141,16 @@ def streamify(arg, mode):
         estream, close_estream = streamify(error,  'w')
         istream, close_istream = streamify(input,  'r')
 
-        quoted_args = [arg for arg in args if re.search(r'^"|^\'|"$|\'$', arg)]
-        if quoted_args:
-            tty.warn(
-                "Quotes in command arguments can confuse scripts like"
-                " configure.",
-                "The following arguments may cause problems when executed:",
-                str("\n".join(["    " + arg for arg in quoted_args])),
-                "Quotes aren't needed because spack doesn't use a shell.",
-                "Consider removing them")
+        if kwargs.get('_warn_quotes', True):
```

Comment:
I would make this public, so use `warn_quotes` and mention it in the docstring.

Reply:
Done.